frontend/src/metabase-lib/lib/queries/StructuredQuery.ts

```diff
@@ -330,28 +330,31 @@ class StructuredQueryInner extends AtomicQuery {
     const sourceQuery = this.sourceQuery();
 
     if (sourceQuery) {
+      const fields = sourceQuery.columns().map(column => {
+        const id = [
+          "field",
+          column.name,
+          {
+            "base-type": column.base_type,
+          },
+        ];
+
+        return new Field({
+          ...column,
+          // TODO FIXME -- Do NOT use field-literal unless you're referring to a native query
```

Comment:
Extracting this block that yields `const fields` into a function with a nice name might further help readers in the future. 

Also smaller cyclomatic complexity.